linera-core/src/node.rs

```diff
@@ -103,6 +102,10 @@ pub trait ValidatorNode {
     /// Subscribes to receiving notifications for a collection of chains.
     async fn subscribe(&self, chains: Vec<ChainId>) -> Result<Self::NotificationStream, NodeError>;
 
+    // Uploads a blob content. Returns an error if the validator has not seen a
+    // certificate using this blob.
+    async fn upload_blob_content(&self, content: BlobContent) -> Result<BlobId, NodeError>;
```

Comment:
Personally, I'd just drop the "content" parts. This is "user-facing" feature and they are thinking about _blobs_ and not really bother w/ the distinction that "blob" has a cached hash etc.